fe/fe-core/src/test/java/com/starrocks/sql/plan/SubqueryTest.java

```diff
@@ -1863,20 +1860,20 @@ public void testCaseWhenSubquery2() throws Exception {
                 "select case when id = abs(0) then 'a' else case when exists (select 1 from tmp) " +
                 "then (select type1 from tmp) > (select pretype from tmp) else null end end from tmp a;";
         String plan = getFragmentPlan(sql);
-        assertContains(plan, "20:Project\n" +
-                "  |  <slot 30> : if(CAST(7: expr AS SMALLINT) = abs(0), 'a', " +
-                "CAST(if(17: expr, 23: expr > 27: expr, NULL) AS VARCHAR))");
+        assertContains(plan, " 19:Project\n" +
+                "  |  <slot 32> : if(8 = abs(0), 'a', CAST(if(17: expr, 24: type1 > 30: pretype, NULL) AS VARCHAR))");
     }
 
-    @Test
-    public void testCaseWhenSubquery3() throws Exception {
-        String sql = "with tmp as (select 8 id, 'season' type1, 'a.season' pretype, 'season' ranktype from dual ) " +
-                "select case when id = abs(0) then 'a' else " +
-                "case when exists (select 1 from tmp) then id in (select id > (select type1 from tmp) from tmp )" +
-                " else null end end from tmp a;";
-        Pair<String, ExecPlan> pair = UtFrameUtils.getPlanAndFragment(connectContext, sql);
-        assertContains(pair.first, "CTEAnchor(cteid=2)");
-    }
+    // we can enable this  test after support constant in sub-query
```

Comment:
the case can't work?

Reply:
> the case can't work?
id in (select id > (select type1 from tmp) from tmp -> 8 in (select id > (select type1 from tmp) from tmp )

Reply:
We need finish this job ASAP.

>  /*
     * @todo: support constant in sub-query
     * e.g.
     *   select v0, 1 in (select v0 from t0) from t1
     *
     * transform to:
     *   select t1.v0,
     *       case
     *           when countRows is null then false;
     *           when v1.t0 is null then null,
     *           when d.v0 is null then null
     *           when d.v0 is not null then true
     *       else false
     *       end
     *   from t1 cross join (
     *       select distinct v0, count(*) as countRows
     *       from t0
     *       where v0 = 1 or v0 is null
     *       order by v0 desc limit 1) d;
     */

Reply:
> We need finish this job ASAP.
> 
> > /*
> > * @todo: support constant in sub-query
> > * e.g.
> > *   select v0, 1 in (select v0 from t0) from t1
> > *
> > * transform to:
> > *   select t1.v0,
> > *       case
> > *           when countRows is null then false;
> > *           when v1.t0 is null then null,
> > *           when d.v0 is null then null
> > *           when d.v0 is not null then true
> > *       else false
> > *       end
> > *   from t1 cross join (
> > *       select distinct v0, count(*) as countRows
> > *       from t0
> > *       where v0 = 1 or v0 is null
> > *       order by v0 desc limit 1) d;
> > */

is this because this optimization make it easier to meet this situation?

Reply:
Yes.